plugins/Installation/HostPortExtractor.php

```diff
@@ -0,0 +1,182 @@
+    {
+        try {
+            if (self::isIPv6($dbHost)) {
+                return self::extractIPv6($dbHost);
+            } elseif (self::isUnixSocket($dbHost)) {
+                return self::extractUnixSocket($dbHost);
+            } elseif (self::isIPWithPort($dbHost)) {
+                return self::extractIPAndPort($dbHost);
+            } else {
+                return null;
+            }
+        } catch (InvalidArgumentException $e) {
+            return null;
+        }
+    }
+
+    /**
+     * Determines if the provided host is correctly formatted IPv6
+     *
+     * @param string $dbHost The user provided database server host
+     * @return boolean Whether the provided host is correct IPv6
+     */
+    private static function isIPv6(string $dbHost): bool
+    {
+        // IPv6 addresses must contain '[' & ']' to be considered valid
+        if (strpos($dbHost, '[') !== false && strpos($dbHost, ']') !== false) {
+            // filter_var requires potential IPv6 addresses to not be encased
+            preg_match_all('/\[(.*?)\]/', $dbHost, $matches);
+            $listOfTextInsideSquareBrackets = $matches[1];
+
+            /*
+             * Only return true if there is some text inside square brackets,
+             * and that text is considered valid IPv6 as per filter_var()
+             */
+            if (
+                count($listOfTextInsideSquareBrackets) > 0 &&
+                filter_var(
+                    $listOfTextInsideSquareBrackets[0],
+                    FILTER_VALIDATE_IP,
+                    FILTER_FLAG_IPV6
+                ) !== false
+            ) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Extracts the Host and Port from a user provided database server host,
+     * assuming the provided host is a IPv6 address, with or without a port
+     *
+     * @param string $dbHost The user provided database server host
+     * @throws InvalidArgumentException if the provided dbHost is not a valid IPv6
+     * @return HostPortExtractor The extracted Host & Port
+     */
+    private static function extractIPv6(string $dbHost): HostPortExtractor
+    {
+        if (!self::isIPv6($dbHost)) {
+            throw new InvalidArgumentException('$dbHost must be a valid IPv6 address');
+        }
+
+        // find and extract the text inside square brackets: []
+        preg_match_all('/\[(.*?)\]/', $dbHost, $matches);
+        $listOfTextInsideSquareBrackets = $matches[1];
+
+        $port = '';
+
+        // extract text after closing bracket to search for port
+        $components = explode(']', $dbHost);
+        if (count($components) > 1) {
+            // check if the text is a valid port e.g. ':3000'
+            preg_match('/^(:\d+)$/', $components[1], $portMatches);
+            if (count($portMatches) > 0) {
+                // only save the port number, not the colon
+                $port = substr($portMatches[1], 1);
+            }
```

Comment:
we don't need to capture the `:`, just the number
```suggestion
            preg_match('/^:(\d+)$/', $components[1], $portMatches);
            if (count($portMatches) > 0) {
                // only save the port number, not the colon
                $port = $portMatches[1];
            }
```

Reply:
The regex is checking to make sure the text retrieved from after the closing square bracket correctly matches the port format. so with "]:3000", it would find ":3000" which is valid, but "]3000" is not a valid port format. 

Though we should probably return null if content is found after the square brackets but it doesn't match the port format, as this indicates an invalid host has been provided.

Reply:
I suggested changing the capturing group, otherwise the regex matches the same thing as before.

Reply:
Oh I see, that makes sense I will make that change.

Reply:
I have addressed this issue.